src/data/geo.tsx

```diff
@@ -0,0 +1,115 @@
+import {get} from '@/api';
+import {logError} from '@/components/Errors';
+import { NearestLookupBody } from '@open-formulieren/formio-renderer/registry/map/types.js';
+import { SearchResult  as LeafletSearchResult, ParseArgument } from 'leaflet-geosearch/src/providers/provider.js';
+import AbstractProvider, { ProviderOptions } from 'leaflet-geosearch/src/providers/provider.js';
+import {useAsync} from 'react-use';
+
+/**
+ * @see `#/components/schemas/GetStreetNameAndCityViewResult` in the API spec.
+ */
+export interface AutoCompleteResult {
+  streetName: string;
+  city: string;
+  secretStreetCity: string;
+}
+
+export const autoCompleteAddress = async (
+  baseUrl: string,
+  postcode: string,
+  houseNumber: string
+): Promise<AutoCompleteResult | null> => {
+  const params: Record<string, string> = {
+    postcode: postcode,
+    house_number: houseNumber,
+  };
+  try {
+    const result = await get<AutoCompleteResult>(`${baseUrl}geo/address-autocomplete`, params);
+    return result!;
+  } catch (error) {
+    logError(error);
+    return null;
+  }
+};
```

Comment:
Note that this was left untouched but is marked as addition because of the file rename,

Edit: Note to self,  the rename can be undone as the file does not contain any jsx syntax anymore.